scripts/neighbor_advertiser

```diff
@@ -356,8 +355,8 @@ def wrapped_ferret_request(request_slice, https_endpoint, http_endpoint):
                                      json=request_slice,
                                      timeout=DEFAULT_REQUEST_TIMEOUT,
                                      verify=False)
-    except ConnectTimeout:
-        log_info("HTTPS Ferret endpoint not found, trying HTTP...")
+    except Exception as e:
```

Comment:
@trzhang-msft @yxieca is it ever possible for the HTTPS endpoint to return a response with an error code (like 401, 403, etc.)? If it is possible, do we want to fail immediately if such an error code is returned or do we want to try the HTTP endpoint?

Reply:
It might be returning these codes. But I think we should still retry until http endpoint also failed. Bypassing http endpoint would be tossing away a chance to work. IMHO.

Reply:
Right, that's my concern. I think if one of those error codes is returned then there won't be an exception, so this method will just return the response to the post_neighbor_advertiser_slice method. That method returns failure if the response code is not 200. So, if we think one of these error codes might be returned by the HTTPS endpoint we might want to move that error handling logic into this method so that we can retry with the HTTP endpoint if need be.

Reply:
I see what you mean. Good catch!

Reply:
@daall is correct. HTTP/HTTPs status codes include non-200, like 400 for bad request. But whether non-200 is treated as an exception is fully client side logic, and I'm not sure about the python built-in behavior here.